### PR TITLE
Adding "matches" statement

### PR DESCRIPTION
**Pull request by jeeyoungk**
grammatically, "describe all MATCHING ..." is more correct, but i'm a
bit conflicted - i don't want to have both MATCHES and MATCHING as
keywords.

WDYT?

### CODE REVIEW

File: query/parser.go [hunk 1]
```diff
@@ -263,24 +259,34 @@ func (p *Parser) makeSelect() {
 	}
 }
 
-func (p *Parser) makeDescribeAll() {
-	p.command = &DescribeAllCommand{}
+func (p *Parser) addNullMatchesClause() {
+	p.pushNode(&matcherClause{regexp.MustCompile("")})
```
Comment (achow) [inline query/parser.go]: `matcherClause{regex:...}`

File: query/parser.go [hunk 2]
```diff
@@ -263,24 +259,34 @@ func (p *Parser) makeSelect() {
...
 }
 
-func (p *Parser) makeDescribeMetrics() {
-	// Pop off the value.
-	stringLiteral, ok := p.popNode(stringLiteralPointer).(*stringLiteral)
+func (p *Parser) addMatchesClause() {
+	compiled := p.popRegex()
+	p.pushNode(&matcherClause{compiled})
```
Comment (achow) [inline query/parser.go]: `matcherClause{regex:...}`

File: query/parser.go [hunk 3]
```diff
@@ -263,24 +259,34 @@ func (p *Parser) makeSelect() {
...
+}
+
+func (p *Parser) makeDescribeAll() {
+	matcherClause, ok := p.popNode(matcherClausePointer).(*matcherClause)
 	if !ok {
 		p.flagTypeAssertion()
 		return
 	}
+	p.command = &DescribeAllCommand{matcherClause}
```
Comment (achow) [inline query/parser.go]: `DescribeAllCommand{matcher:...}`

**Comment by achow** [comment]
I think we have a few alternatives:

- Use `match`. It's more of an imperative, but I think it's kind of fits?
- Use `like`, like SQL.
- Cheese it with supporting pipes. We can `describe all | grep`.

Otherwise LGTM after nits.

**Comment by jeeyoungk** [comment]
@achow

* nits fixed.
* matches -> match.

PTAL

**Comment by Nathan-Fenner** [comment]
LGTM

**Comment by syamp** [comment]
I think "like" is reads better.

**Pull request merged**